### DISCUSSION TITLE
About using efr32-radio in contiki

### DISCUSSION THREAD

**Discussion by Thanhnt-64** [contiki-ng/contiki-ng#2817] [Q&A]
I'm porting MGM240PB32VNN (DEVICE_FAMILY := MGM24   |   RAIL_CPU := efr32xg24   |   RAIL_FAMILY := efr32xg2x)
I had a little trouble porting the efr32-radio driver: in efr32-radio.c (init function)
![Capture](https://github.com/contiki-ng/contiki-ng/assets/92809811/d70f7ffd-503b-46a8-a73d-9825c6edb634)
When I use two functions: RAIL_ConfigChannels() and RAIL_IEEE802154_Config2p4GHzRadio() like the above image, my program crashes.
If I remove those two functions, the program initializes and runs normally. But they are important initializations in network configuration. Can you help me solve this problem?

**Comment by Yagoor** [comment]
Can you narrow down which of the two functions cause the hard fault?
These functions are hard to debug because they are in the rail library, which we don't have access to the source code.

**Reply by Thanhnt-64** [reply]
I tried deleting each of these two functions but still got the error. Only when deleting both functions will the program error disappear. Do you think it could be due to an error/conflict somewhere?

**Reply by Thanhnt-64** [reply]
the rail lib in SDK:  https://github.com/SiliconLabs/gecko_sdk

**Reply by Yagoor** [reply]
I honestly would need to debug it myself.
I have checked and there are some boards with the MG24 supported by Renode, so I can take a look at it without the actual hardware.
However, I don't know when I'll have time for it myself.

**Comment by Yagoor** [comment]
Which board do you have? I found a old branch in which I was adding support for the brd4184 and I extended it to the brd2601b (which has an MG24 OPN)
https://github.com/contiki-ng/contiki-ng/pull/2826
This is not tested at all, and I don't know when I will find time for it.

**Reply by khanhongithub** [reply]
We dont use any dev kit, we are porting directly onto the module mgm240p
